lib/srv/ingress/reporter.go

```diff
@@ -87,17 +88,50 @@ var (
 // HTTPConnStateReporter returns a http connection event handler function to track
 // connection metrics for an http server.
 func HTTPConnStateReporter(service string, r *Reporter) func(net.Conn, http.ConnState) {
+	tracker := conntrack{
```

Comment:
Have you considered using [`sync.Map`](https://pkg.go.dev/sync#Map)? It might be just what you need with [`LoadAndDelete`](https://pkg.go.dev/sync#Map.LoadAndDelete) and [`LoadOrStore`](https://pkg.go.dev/sync#Map.LoadOrStore).

Reply:
Switched to `sync.Map` here [c92fa17](https://github.com/gravitational/teleport/pull/24058/commits/c92fa170854870ea23e97b71cc69b0ee2e2e7f73). And deleted the conn tracker here [de4c56d](https://github.com/gravitational/teleport/pull/24058/commits/de4c56ddebfae9085605ecf5126a5be0e73d8151)